### ISSUE TITLE
"Your extension just crashed with a bug that we have been trying to replicate for a long time"

### ISSUE THREAD

**Issue by jlbosse** [julia-vscode/LanguageServer.jl#1058]
The LSP under neovim just crashed on me with the message "LPS[julials] Julia Extension: Please contact us! Your extension just crashed with a bug that we have been trying to replicate for a long time.".

I suppose you will need me to upload some log files? Where do I find them? And what other information do you need?

**Comment by josephsdavid** [comment]
I am having the same issue! It seems to happen to me when I run the formatting when editing a remote file through SCP

**Comment by pxl-th** [comment]
After some recent updates to LanguageServer I get this error all the time.
I think it started happening after #1105, because I reverted to before that (ddcbc1ac2cea5a096c3390273601337c823145a2) and this error does not appear.

Previously this didn't happen at all and I haven't changed NeoVim version.

```julia
ERROR:
    Mismatch between server and client text for file:///home/pxl-th/code/NerfModels.jl/examples/train.jl. _open_in_editor is true. _workspace_file is true. _version is 236.
Stacktrace:
[1] textDocument_didSave_notification(params::LanguageServer.DidSaveTextDocumentParams, server::LanguageServerInstance, conn::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint})
    @ LanguageServer ~/.julia/packages/LanguageServer/oAWgk/src/requests/textdocument.jl:50
[2] (::LanguageServer.var"#98#99"{typeof(LanguageServer.textDocument_didSave_notification), LanguageServerInstance})(conn::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint}, params::LanguageServer.DidSaveTextDocumentParams)
    @ LanguageServer ~/.julia/packages/LanguageServer/oAWgk/src/languageserverinstance.jl:262
[3] dispatch_msg(x::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint}, dispatcher::JSONRPC.MsgDispatcher, msg::Dict{String, Any})
    @ JSONRPC ~/.julia/packages/JSONRPC/P0G1p/src/typed.jl:67
[4] run(server::LanguageServerInstance)
    @ LanguageServer ~/.julia/packages/LanguageServer/oAWgk/src/languageserverinstance.jl:382
[5] runserver(pipe_in::Base.PipeEndpoint, pipe_out::Base.PipeEndpoint, env_path::String, depot_path::String, err_handler::Nothing, symserver_store_path::Nothing)
    @ LanguageServer ~/.julia/packages/LanguageServer/oAWgk/src/runserver.jl:41
[6] runserver()
    @ LanguageServer ~/.julia/packages/LanguageServer/oAWgk/src/runserver.jl:39
[7] top-level scope
    @ none:1
```

NeoVim version:

```
NVIM v0.7.0
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by runner@fv-az316-460
```

**Comment by fredrikekre** [comment]
I am also running with ddcbc1ac2cea5a096c3390273601337c823145a2. @pfitzseb believes this is a problem with neovim, but I have not had time to narrow it down.

**Comment by fredrikekre** [comment]
I made some more debugging here and I see two different versions of this crash;

1. Sometimes I get a crash on first save no matter what I do. This happens because in those sessions neovim doesn't send *any* `textDocument/didChange` notifications, so when saving it errors. This must clearly be a bug in neovim, but I have not yet investigated whether that is only happening with incremental sync (https://github.com/julia-vscode/LanguageServer.jl/pull/1105). Trace here: https://gist.github.com/fredrikekre/d917350da5f8b3fbc89f3092964c6db2#file-lsp-no-didchange-log

2. It also sometimes crashes even though it has been sending the change notification, but I have not yet found the exact cause, but here is a trace: https://gist.github.com/fredrikekre/d917350da5f8b3fbc89f3092964c6db2#file-lsp-didchange-log

**Comment by mfiano** [comment]
This is still an issue quite frequently but hard to reproduce reliably, using LS.jl 4.3.1 on Neovim 0.7.2:

```
[ERROR][2022-07-11 02:49:16] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"ERROR: "
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"Mismatch between server and client text for file:///home/mfiano/projects/Julia/Practice/src/Practice.jl. _open_in_editor is true. _workspace_file is true. _version is 0.\nStacktrace:"
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"\n [1] textDocument_didSave_notification(params::"
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"LanguageServer.DidSaveTextDocumentParams"
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	", server::LanguageServerInstance, conn::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint})\n   @ LanguageServer "
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"~/.data/julia/packages/LanguageServer/0vsx2/src/requests/"
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"textdocument.jl:50\n [2] "
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	'(::LanguageServer.var"#98#99"{typeof(LanguageServer.textDocument_didSave_notification), LanguageServerInstance})('
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"conn::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint}, params::LanguageServer.DidSaveTextDocumentParams)\n   @ LanguageServer "
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"~/.data/julia/packages/LanguageServer/0vsx2/src/languageserverinstance.jl:262\n "
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"[3] dispatch_msg(x::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint}, dispatcher::JSONRPC.MsgDispatcher"
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	", msg::"
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"Dict{String, Any})\n   @ "
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"JSONRPC ~/.data/julia/packages/JSONRPC/P0G1p/src/typed.jl:67\n [4]"
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	" run(server::LanguageServerInstance)\n   @ LanguageServer "
[ERROR][2022-07-11 02:49:17] .../vim/lsp/rpc.lua:420	"rpc"	"julia"	"stderr"	"~/.data/julia/packages/LanguageServer/0vsx2/src/languageserverinstance.jl:382\n [5] top-level scope\n   @ none:31\n"
```

**Comment by fredrikekre** [comment]
Since I don't see this anymore I can only assume it is related to your configuration somehow. Would be great if you can open a new issue with some minimal config and steps to reproduce, or at least enable verbose logging and submit that (`vim.lsp.log.set_level("TRACE")` in your rc file)

**Comment by jw3126** [comment]
> using LS.jl 4.3.1 on Neovim 0.7.2:

is it fixed in 4.3.1 or do you need master?

**Comment by pfitzseb** [comment]
> is it fixed in 4.3.1

Yes.

**Comment by mfiano** [comment]
> (`vim.lsp.log.set_level("TRACE")` in your rc file

Note to anyone following along on Neovim 0.7.2: It's actually: `*vim.lsp.set_log_level("TRACE")`
